### DISCUSSION TITLE
V4L2 / Video Capture Support on PolarFire-SoC Linux

### DISCUSSION THREAD

**Discussion by mcmordie** [polarfire-soc/polarfire-soc-documentation#103] [Linux drivers for PolarFire Soc] [answered]
Good afternoon,

I am designing a video capture platform around the MPFS250.  I have RTL to capture video over MIPI CSI2 and stream it into main memory, but how to support this from inside Linux?  Are there any Video4Linux2 (V4L2) capture drivers and examples available or planned for this or related platforms?

What is the best way to approach sharing memory between RTL blocks and linux?  Can we simply write into a particular address block from the RTL side and then read that memory from the linux side?

Thanks,

Dave

**Comment by nitindeshpande** [comment, marked as answer]
@mcmordie Examples and Drivers for Video related use-cases are planned on our 
[PolarFire SoC Video Kit](https://www.microchip.com/en-us/development-tool/MPFS250-VIDEO-KIT) platform in the next release.

You may refer some of the relevant documentation [here](https://www.microchip.com/en-us/products/fpgas-and-plds/system-on-chip-fpgas/polarfire-soc-fpgas#Documentation).

**Reply by mcmordie** [reply]
Thank you @nitindeshpande
